### DISCUSSION TITLE
How to resume a bookmark in an activity using Api / curl

### DISCUSSION THREAD

**Discussion by saisworld** [elsa-workflows/elsa-core#5082] [Q&A]
I have a custom activity and added bookmark. When I trigger the Workflow its correctly waiting for some event to resume. How do I resume the Activity using Api call for eg using curl statement.?

**Comment by glime-ai** [comment]
To resume a workflow activity that is waiting on a bookmark using an API call, you can use the `RunWorkflowOptions` with the bookmark ID to resume the workflow. Based on the context provided, you would typically use a higher-level service such as `IWorkflowRuntime` to resume workflows when implementing custom blocking activities. However, if you want to resume the workflow using an API call, you would need to make a POST request to the appropriate endpoint with the necessary payload.

Here's an example of how you might construct a curl command to resume a workflow:

```bash
curl --location --request POST 'https://localhost:5001/v1/workflows/{workflow_instance_id}/dispatch' \
--header 'Content-Type: application/json' \
--data-raw '{
    "BookmarkId": "{bookmark_id}",
    "Input": {
        // Any additional input data required for the workflow to resume
    }
}'
```

In this example, replace `{workflow_instance_id}` with the ID of the workflow instance you want to resume and `{bookmark_id}` with the ID of the bookmark that was created by the custom activity. The `Input` field is optional and should contain any data that the workflow expects when it resumes.

Please note that the exact endpoint and payload structure may vary depending on the version of Elsa Workflow you are using and how the workflow and custom activity are configured. If you are using Elsa 3.0 or later, you might need to trigger an event instead, as shown in the GitHub issue example:

```bash
curl --location --request POST 'https://localhost:7197/elsa/api/events/{event_name}/trigger' \
--header 'Content-Type: application/json' \
--data-raw '{
    "BookmarkId": "{bookmark_id}",
    "Input": {
        // Any additional input data required for the workflow to resume
    }
}'
```

Replace `{event_name}` with the name of the event that the custom activity is waiting for, and `{bookmark_id}` with the ID of the bookmark.

If you are unsure about the exact endpoint or payload, you should refer to the Elsa Workflow documentation or the API definition for the version you are using.

**Reply by saisworld** [reply]
How to do this in Elsa V3?

**Reply by saisworld** [reply]
In the customActivity, all I have is -   `var bookmark = context.CreateBookmark();` 

Where do I need to define the Event name?
